apps/web/pages/api/book/event.ts

```diff
@@ -202,23 +228,15 @@ const getEventTypesFromDB = async (eventTypeId: number) => {
 
 type User = Prisma.UserGetPayload<typeof userSelect>;
 
-type ExtendedBookingCreateBody = BookingCreateBody & {
-  noEmail?: boolean;
-  recurringCount?: number;
-  rescheduleReason?: string;
-};
-
-export default async function handler(req: NextApiRequest, res: NextApiResponse) {
-  const { recurringCount, noEmail, ...reqBody } = req.body as ExtendedBookingCreateBody;
+async function handler(req: NextApiRequest) {
+  const { recurringCount, noEmail, eventTypeSlug, eventTypeId, hasHashedBookingLink, language, ...reqBody } =
+    extendedBookingCreateBody.parse(req.body);
```

Comment:
[Validate using zod instead of type casting](https://app.gitbook.com/o/6snd8PyPYMhg0wUw6CeQ/s/VXRprBTuMlihk37NQgUU/engineering/best-practices/validate-using-zod-instead-of-type-casting)